src/main/java/com/example/solidconnection/university/repository/UnivApplyInfoRepository.java

```diff
@@ -10,40 +9,50 @@
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 import static com.example.solidconnection.common.exception.ErrorCode.UNIV_APPLY_INFO_NOT_FOUND;
 
 @Repository
 public interface UnivApplyInfoRepository extends JpaRepository<UnivApplyInfo, Long>, UnivApplyInfoFilterRepository {
 
     @Query("""
-            SELECT uai
-            FROM UnivApplyInfo uai
-            JOIN University u ON uai.university = u
-            WHERE (u.country.code IN (
-                      SELECT c.code
-                      FROM InterestedCountry ic
-                      JOIN ic.country c
-                      WHERE ic.siteUser = :siteUser
-                  )
-                  OR u.region.code IN (
-                      SELECT r.code
-                      FROM InterestedRegion ir
-                      JOIN ir.region r
-                      WHERE ir.siteUser = :siteUser
-                  ))
-                  AND uai.term = :term
-            """)
-    List<UnivApplyInfo> findAllBySiteUsersInterestedCountryOrRegionAndTerm(@Param("siteUser") SiteUser siteUser, @Param("term") String term);
+        SELECT DISTINCT uai
+        FROM UnivApplyInfo uai
+        LEFT JOIN FETCH uai.languageRequirements lr
+        JOIN FETCH uai.university u
+        LEFT JOIN FETCH u.country c
+        LEFT JOIN FETCH u.region r
+        WHERE (c.code IN (
+                  SELECT ic.countryCode
+                  FROM InterestedCountry ic
+                  WHERE ic.siteUserId = :siteUserId
+              )
+              OR r.code IN (
+                  SELECT ir.regionCode
+                  FROM InterestedRegion ir
+                  WHERE ir.siteUserId = :siteUserId
+              ))
+              AND uai.term = :term
+    """)
+    List<UnivApplyInfo> findAllBySiteUsersInterestedCountryOrRegionAndTerm(@Param("siteUserId") Long siteUserId, @Param("term") String term);
 
-    @Query(value = """
-                SELECT *
-                FROM university_info_for_apply
-                WHERE term = :term
-                ORDER BY RAND()
-                LIMIT :limitNum
-            """, nativeQuery = true)
-    List<UnivApplyInfo> findRandomByTerm(@Param("term") String term, @Param("limitNum") int limitNum);
+    @Query("""
+    SELECT DISTINCT uai
+    FROM UnivApplyInfo uai
+    LEFT JOIN FETCH uai.languageRequirements lr
+    LEFT JOIN FETCH uai.university u
+    LEFT JOIN FETCH u.country c
+    LEFT JOIN FETCH u.region r
+    WHERE uai.term = :term
+    ORDER BY FUNCTION('RAND')
+    """)
```

Comment:
```suggestion
    @Query("""
            SELECT DISTINCT uai
            FROM UnivApplyInfo uai
            LEFT JOIN FETCH uai.languageRequirements lr
            LEFT JOIN FETCH uai.university u
            LEFT JOIN FETCH u.country c
            LEFT JOIN FETCH u.region r
            WHERE uai.term = :term
            ORDER BY FUNCTION('RAND')
            """)
```

개인적으로 코드 내부의 개행은 개발자의 자유라고 생각하는데, 하나의 클래스에서 이미 유지하고 있는 방식이 있다면 그걸로 통일하는게 좋다 생각합니다 😊

Reply:
수정하였습니다!